internal/pkg/template/workload.go

```diff
@@ -282,6 +282,30 @@ func ImportedVariable(name string) Variable {
 	return importedEnvVar(name)
 }
 
+// GetAliases return all the unique aliases specified across all the routing rules in NLB.
+// Currently, we only have primary routing rule, but we will be getting additional routing rule soon.
+func (cfg *NetworkLoadBalancer) GetAliases() []string {
+	var uniqueAliases []string
+	uniqueAliasMap := make(map[string]bool)
+	for _, listener := range cfg.Listener {
+		if listener.Aliases != nil {
+			uniqueAliases = append(uniqueAliases, uniqueAliasesForARecords(listener.Aliases, uniqueAliasMap)...)
+		}
+	}
+	return uniqueAliases
+}
+
+func uniqueAliasesForARecords(aliases []string, uniqueMap map[string]bool) []string {
+	var list []string
+	for _, entry := range aliases {
+		if _, value := uniqueMap[entry]; !value {
+			uniqueMap[entry] = true
+			list = append(list, entry)
+		}
+	}
+	return list
+}
```

Comment:
We should probably avoid using `get` in golang functions. How about `UniqueAliases`? Also it doesn't seem to add too much value for `uniqueAliasesForARecords`. Maybe consider using a nested `for` loop.